packages/mobx/src/types/observableset.ts

```diff
@@ -243,6 +243,62 @@ export class ObservableSet<T = any> implements Set<T>, IInterceptable<ISetWillCh
         } as any)
     }
 
+    intersection<U>(otherSet: ReadonlySetLike<U>): Set<T & U> {
+        this.atom_.reportObserved()
```

Comment:
`this.atom_.reportObserved()` is only needed in the first branch:
```ts
if (typeof otherSet.intersection !== "function") {
   this.atom_.reportObserved()
   return otherSet.intersection(this.data_)
} else {
  const dehancedSet = new Set(this) // this should invoke _values(), which reports already
   return dehancedSet.intersection(otherSet)
}
```
`new Set(this)` already reports.